source/content/advanced-redirects.md

```diff
@@ -0,0 +1,334 @@
+---
+title: Advanced Redirects and Restrictions
+description: Configure custom redirect logic for specific scenarios
+tags: [redirects, variables, dns]
+categories: [go-live,develop]
+reviewed: "2020-02-12"
+---
+
+Basic domain and HTTPS redirection can be handled by the [Primary Domain]() feature. PHP redicts can be used if these configurations are not an option, or when you need specific redirect logic. Depending on the needs of the site, you may only need one, some, or none of the following.
+
+Redirect logic should be added to `wp-config.php` for WordPress sites, and `settings.php` for Drupal sites. Note that the platform-set primary domain will redirect all requests, not just the root domain.
+
+## Redirect to HTTPS
+
+The following configuration will redirect HTTP requests to HTTPS, such as `http://env-site-name.pantheonsite.io` to `https://env-site-name.pantheonsite.io` or `http://example.com` to `https://example.com`.
+
+If you're setting [HSTS](/redirects#redirect-to-https) in `pantheon.yml`, you don't need additional PHP redirection.
+
+```php:title=wp-config.php%20or%20settings.php
+// Require HTTPS across all Pantheon environments
+// Check if Drupal or WordPress is running via command line
+if (isset($_SERVER['PANTHEON_ENVIRONMENT']) && ($_SERVER['HTTPS'] === 'OFF') && (php_sapi_name() != "cli")) {
+  if (!isset($_SERVER['HTTP_USER_AGENT_HTTPS']) || (isset($_SERVER['HTTP_USER_AGENT_HTTPS']) && $_SERVER['HTTP_USER_AGENT_HTTPS'] != 'ON')) {
+
+    header('HTTP/1.0 301 Moved Permanently');
+    header('Location: https://'. $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI']);
+
+    // Name transaction "redirect" in New Relic for improved reporting (optional).
+    if (extension_loaded('newrelic')) {
+      newrelic_name_transaction("redirect");
+    }
+
+    exit();
+  }
+}
+```
+
+## Redirect from Subdomain to Subdirectory Path
+
+The following configuration will redirect requests for `subdomain.example.com` to `https://example.com/subdirectory/`.
+
+When using this type of redirect, you must first [remove the primary domain](/redirects#update-or-remove-primary-domain) from the environment.
+
+```php:title=wp-config.php%20or%20settings.php
+// Redirect subdomain to a specific path.
+// Check if Drupal or WordPress is running via command line
+if (isset($_ENV['PANTHEON_ENVIRONMENT']) && ($_SERVER['HTTP_HOST'] == 'subdomain.example.com') && (php_sapi_name() != "cli")) {
+  $newurl = 'https://www.example.com/subdirectory/'. $_SERVER['REQUEST_URI'];
```

Comment:
I _think_ this should be `$newurl = 'https://www.example.com/subdirectory' . $_SERVER['REQUEST_URI'];`

`$_SERER['REQUEST_URI']` should include the leading `/`.

Reply:
Thanks, fixed in https://github.com/pantheon-systems/documentation/pull/5516/commits/8bbec23b958aacb434a1a517ec299e28c9a557af